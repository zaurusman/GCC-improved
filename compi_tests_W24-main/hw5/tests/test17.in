    int player1;
    int player2;

    print("Player 1, enter any number: ");
    player1 = readi(0);
    printi(player1);

    int divisors1 = 0;
    int divisors2 = 0;

    int turn = 2; // Player 2 starts the game
    while (true) {
    
        // Count divisors for player1's number
        int j = 1;
        while (j <= player1) {

            if (player1 / j * j == player1) {
                divisors1 = divisors1 + 1;
            }
            j = j + 1;
        }

        if (turn == 2) {
            print("Player 2, enter a number larger than %d with the same number of divisors: ");
            player2 = readi(0);
            printi(player2);
            
            // Count divisors for player2's number
            divisors2 = 0;
            int i = 1;
            while (i <= player2) {
                if (player2 / i * i == player2) {
                    divisors2 = divisors2 + 1;
                }
                i = i + 1;
            }

            if (player2 <= player1 or divisors1 != divisors2) {
                print("Player 2 loses!");
                break;
            }
        } else {
            print("Player 1, enter a number larger than %d with the same number of divisors: ");
            player1 = readi(0);
            printi(player1);

            // Count divisors for player1's number
            divisors1 = 0;
            int i = 1;
            while (i <= player1) {

                if (player1 / i * i == player1) {
                    divisors1 = divisors1 + 1;
                }
                i = i + 1;
            }

            if (player1 <= player2 or divisors1 != divisors2) {
                print("Player 1 loses!");
                break;
            }
        }
        turn = 3 - turn; // Switch turns between players (1 -> 2 or 2 -> 1)
    }
    print("Player 1 divisors:");
    printi(divisors1);
    print("Player 2 divisors:");
    printi(divisors2);