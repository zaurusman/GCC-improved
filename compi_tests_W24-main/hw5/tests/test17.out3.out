Player 1, enter any number: 
60
Player 2, enter a number larger than %d with the same number of divisors: 
90
Player 1, enter a number larger than %d with the same number of divisors: 
150
Player 2, enter a number larger than %d with the same number of divisors: 
1
Player 2 loses!
Player 1 divisors:
24
Player 2 divisors:
1
